src/main/java/net/minecraftforge/common/util/LazyOptional.java

```diff
@@ -108,18 +108,13 @@ private LazyOptional(@Nullable NonNullSupplier<T> instanceSupplier)
         if (supplier != null)
         {
             resolved = new AtomicReference<>(null);
-            try
+            T temp = supplier.get();
+            if (temp == null)
             {
-                T temp = supplier.get();
-                if (temp == null)
-                    throw new IllegalStateException("Supplier must not return null value");
-                resolved.set(temp);
-                return resolved.get();
-            }
-            catch (Throwable e)
-            {
-                return null;
+                throw new NullPointerException("Supplier should not return null value");
```

Comment:
We do not want to throw an exception to be caught by the consumer. Logging it noisily is probably the way to go.

Reply:
I've reverted back to the previous version that does that.